### PR TITLE
WORKDIR is like calling mkdir - but we've not told people

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit SvenDowideit@home.org.au

### CODE REVIEW

File: docs/reference/builder.md [hunk 1]
```diff
@@ -1045,6 +1045,9 @@ For example:
 The output of the final `pwd` command in this `Dockerfile` would be
 `/path/$DIRNAME`
 
+Each `WORKDIR` will be created even if its not used in any subsequent `Dockerfile`
+instruction.
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Perhaps include it in the first paragraph; https://github.com/docker/docker/pull/21340/files#diff-4fc21624aa5dadd153e2780910da7c7dR1023, e.g. "If the workdir doesn't exist, ..."?

**Comment by jhowardmsft** [comment]
@swernli. Hmmm this might not quite be true on Windows. I don't believe we can create the directory in the case of Hyper-V containers. Can check tomorrow when in office, unless Stefan can confirm.

**Comment by swernli** [comment]
Yes, John is correct here: there is a limitation in HYPER-V containers currently where the daemon currently cannot create folders directly in the container storage. So the code that creates the WORKDIR folder on each container start is skipped for HYPER-V containers.  However, we consider this a bug (and are tracking it as such internally) and are hoping to fix it so that the behavior for HYPER-V containers is the same as others.  Given that, I think it's ok to make the behavior of setting a WORKDIR clearer in the documentation.

**Comment by SvenDowideit** [comment]
@thaJeztah updated :)

**Comment by thaJeztah** [comment]
thanks for looking @swernli 

LGTM

**Comment by chrisfosterelli** [comment]
Can anyone comment on how long WORKDIR has had this behaviour, did it always act like mkdir? I noticed it just appeared in the docs for 1.10 but if we expect `WORKDIR` to create the directory will it cause issues for non-1.10 users? (cc @SvenDowideit)

**Pull request merged**